internal/util/sessionutil/session_util.go

```diff
@@ -487,3 +505,96 @@ func (s *Session) Registered() bool {
 	}
 	return b
 }
+
+func (s *Session) SetEnableActiveStandBy(enable bool) {
+	s.enableActiveStandBy = enable
+}
+
+func (s *Session) updateStandby(b bool) {
+	s.isStandby.Store(b)
+}
+
+func (s *Session) ProcessActiveStandBy(activateFunc func()) error {
+	if !s.enableActiveStandBy {
+		return nil
+	}
+	key := s.ServerName
+	primaryLockKey := path.Join(s.metaRoot, DefaultServiceRoot+"primaryLock/", key)
+	completeKey := path.Join(s.metaRoot, DefaultServiceRoot, s.ServerName)
+
+	// lock for the key to ensure only one active
+	lockFn := func() error {
+		// this ttl must no less than the TTL for register lease
+		session, err := concurrency.NewSession(s.etcdCli, concurrency.WithTTL(DefaultTTL))
+		if err != nil {
+			log.Debug("Fail to create session", zap.Error(err))
+		}
+		//defer session.Close()
+		log.Debug("Try get lock for key", zap.Int64("ServerID", s.ServerID))
+		s.activeLock = concurrency.NewMutex(session, primaryLockKey)
+		// acquire lock for session
+		err = s.activeLock.TryLock(s.ctx)
+		if err != nil {
+			log.Debug("Try lock error", zap.Error(err))
+			session.Close()
```

Comment:
What about other paths? Can session close properly if err == nil?

Reply:
The session will always be kept when the coord is active. if the session closes, the lock will release too.

Reply:
Ah nvm. I thought we need to explicitly call Close() at some time point for cleaning up. It can close itself after TTL.

Altho, do we need to keep this session open? Would the session expire after TTL=60s?

Reply:
Currently it is needed to kept open and won't expire while the lock exists. I will think about if there is another graceful impl

Reply:
I have adjusted the implement. Use lease instead of session and mutex to lock the primary node key, the lease will be handled by keepAlive logic.